File: backend/golang/models/customer.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Customer struct {
	ID        primitive.ObjectID `json:"id" bson:"_id,omitempty" required:"true"`
	FirstName string             `json:"first_name" bson:"first_name" required:"true"`
	LastName  string             `json:"last_name" bson:"last_name" required:"true"`
	Phone     string             `json:"phone" bson:"phone" required:"true"`
	Age       *int                `json:"age" bson:"age" required:"true"`
	Sex       *bool               `json:"sex" bson:"sex" required:"true"`
	CreatedAt time.Time          `json:"createdAt,omitempty" bson:"createdAt,omitempty"`
	UpdatedAt time.Time          `json:"updatedAt,omitempty" bson:"updatedAt,omitempty"`
}

File: frontend/thunder-opto-next-js/models/customer.ts
// export type Customer = {
//     id: string;
//     firstName: string;
//     lastName: string;
//     phone: string;
//     age: number;
//     sex: boolean;
// }

export interface Customer {
  id: string;
  first_name: string;
  last_name: string;
  phone: string;
  age: number;
  sex: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface NewCustomer {
  first_name: string;
  last_name: string;
  phone: string;
  age: number;
  sex: boolean;
}